File: 0_algo/Ch04/go/array.go
package main

import (
	"fmt"
	"math/rand"
)

// 随机访问元素
func randomAccess(arr []int) int {
	idx := rand.Intn(len(arr))
	return arr[idx]
}

// 插入元素
func insert(arr []int, num int, idx int) {
	for i := len(arr) - 1; i > idx; i-- {
		arr[i] = arr[i-1]
	}

	arr[idx] = num
}

// 删除元素
func delete(arr []int, idx int) {
	for i := idx; i < len(arr)-1; i++ {
		arr[i] = arr[i+1]
	}
}

// 查找元素
func find(arr []int, num int) bool {
	for i := range arr {
		if i == num {
			return true
		}
	}

	return false
}

func main() {
	arr := []int{1, 2, 3, 4, 5, 6, 7, 8, 9}
	num1 := randomAccess(arr)
	fmt.Println(num1)

	insert(arr, 100, 3)
	fmt.Println(arr)

	delete(arr, 3)
	fmt.Println(arr)

	res := find(arr, 10)
	fmt.Println(res)

}

File: 0_algo/Ch04/array.cpp
#include <iostream>

using namespace std;

void print_arr(int arr[], int size)
{
    for (int i = 0; i < size; ++i)
    {
        cout << *arr++ << " ";
    }
    cout << endl;
}

// 随机访问
int random_access(int arr[], int size)
{
    int i = rand() % size;

    return arr[i];
}

// 插入元素
void insert(int arr[], int size, int idx, int num)
{
    for (int i = size - 1; i > idx; --i)
    {
        arr[i] = arr[i-1];
    }

    arr[idx] = num;
}

int main()
{
    int arr[5];
    int nums[5] = {1, 2, 3, 4, 5};

    int num = random_access(nums, 4);
    cout << num << endl;
    print_arr(nums, 5);

    insert(nums, 5, 2, 100);
    print_arr(nums, 5);
}